### ISSUE TITLE
Deriving loss terms

### ISSUE THREAD

**Issue by QEDan** [XiaoyuBIE1994/DVAE#3]
I'm hoping to adapt the DVAEs from here to another domain. But, I have a question about the loss terms. Take the SRNN for example. The reconstruction loss in the code is given by this:

`loss_recon = torch.sum( x/y - torch.log(x/y) - 1)`

This loss corresponds to equation (9.15) of your review article:
![image](https://user-images.githubusercontent.com/9982903/124525208-3dc2e780-ddb3-11eb-9dc9-bf33fd0aa5a2.png)

For a squared magnitude spectrogram, x and y are never negative, but for other domains they could be negative and this reconstruction loss must take a different form to avoid taking the log of negative numbers. I don't see how to derive the reconstruction loss used in the code from the general form given in the paper. Is this math worked out in any of the references? If not, would you mind outlining the steps to get from one to the other?

**Comment by QEDan** [comment]
After reading the review paper (https://arxiv.org/abs/2008.12595) more thoroughly, I think I can answer my question. 

In section 13.2.1 of the review article associated with this repo, the authors explain how the STFT coefficients and power spectrogram are modelled:
`Modeling the STFT coefficients with a zero-mean circular complex Gaussian distribution ... amounts to modeling each entry of the power spectrogram with a Gamma distribution... This also amounts to using the Itakura-Saito distance ... in the reconstruction term of the VLB, see (Girin et al., 2019).`

The referenced paper (Girin et al., 2019) can be found here: http://www.gipsa-lab.fr/~laurent.girin/papers/Girin_et_al_DAFx2019.pdf
This paper explains in greater detail the relationship between the choice of model of the spectrogram and the corresponding form of the reconstruction loss term. In the case of their choice of spectrogram model, the resulting loss is the [Itakura-Saito divergence](https://en.wikipedia.org/wiki/Itakura%E2%80%93Saito_distance), which is exactly the form used in their code.

Girin et al. also explain that for VAEs in domains with negative numbers in the inputs, the Itakura-Saito loss would not be appropriate, but other choices might be such as a KL-divergence or Euclidean distance. Reading that article will help to clarify the relationship between the reconstruction loss and the corresponding model of the data generating process.

**Comment by XiaoyuBIE1994** [comment]
Great! Thx for reading our paper, this is exactly what it means in the code. Actually, the formulation of log-likelihood depends on which kind of distribution you assume for your data. Normally a Gaussian should be OK, but not work for the speech spectrum.

**Comment by Sakura-gh** [comment]
Thanks for your explain! so the loss term implemented in code for reconstruction error should not be deterministic form, but be the suitable form for what kind of the data we use.
